src/ansys/dpf/post/simulation.py

```diff
@@ -0,0 +1,524 @@
+            + (selection is not None)
+        )
+        if tot > 1:
+            raise ValueError(
+                "Arguments selection, named_selections, element_ids, "
+                "and node_ids are mutually exclusive"
+            )
+        if selection is not None:
+            return selection
+        else:
+            selection = Selection(server=self._model._server)
+        # Create the SpatialSelection
+        if named_selections:
+            selection.select_named_selection(named_selection=named_selections)
+        elif element_ids:
+            if location == locations.nodal:
+                selection.select_nodes_of_elements(elements=element_ids, mesh=self.mesh)
+            else:
+                selection.select_elements(elements=element_ids)
+        elif node_ids:
+            if location != locations.nodal:
+                raise ValueError(
+                    "Argument 'node_ids' can only be used if 'location' "
+                    "is equal to 'post.locations.nodal'."
+                )
+            selection.select_nodes(nodes=node_ids)
+        # Create the TimeFreqSelection
+        if all_sets:
+            selection.time_freq_selection.select_with_scoping(
+                dpf.time_freq_scoping_factory.scoping_on_all_time_freqs(self._model)
+            )
+        elif set_ids is not None:
+            if isinstance(set_ids, int):
+                set_ids = [set_ids]
+            selection.select_time_freq_sets(time_freq_sets=set_ids)
+
+        elif times is not None:
+            # Check input
+            if isinstance(times, list):
+                if any([not (type(t) in [float, int]) for t in times]):
+                    raise ValueError("Argument times must contain numeric values only.")
+            elif isinstance(times, float) or isinstance(times, int):
+                times = [times]
+            else:
+                raise TypeError("Argument times must be a number or a list of numbers.")
+
+            # Get the set_ids for available time values matching the requested time values.
+            available_times = self.time_freq_support.time_frequencies.data
+            precision = self._get_time_freq_precision
+            available_times_to_extract_set_ids = []
+            last_extracted_index = -1
+            len_available = len(available_times)
+            for t in times:
+                found = False
+                i = last_extracted_index + 1
+                while i < len_available:
+                    if abs(float(t) - available_times[i]) < precision:
+                        last_extracted_index = i
+                        available_times_to_extract_set_ids.append(i + 1)
+                        found = True
+                    i += 1
+                if not found:
+                    raise ValueError(
+                        f"Could not find time={t}{self.units['time/frequency']} "
+                        f"in the simulation."
```

Comment:
I would probably use ``np.round`` for precision and then ``np.isin``.

https://numpy.org/doc/stable/reference/generated/numpy.isin.html
https://stackoverflow.com/a/63949164

Not sure if this might help

Reply:
I'll keep it as is for now since dpf-core is apparently able to do this if done correctly. This is among the first refactors I will be doing (DPF operators actually allow for interpolation).